src/main/java/org/broadinstitute/hellbender/tools/copynumber/allelic/alleliccount/AllelicCountCollection.java

```diff
@@ -19,73 +19,43 @@
  * @author Samuel Lee &lt;slee@broadinstitute.org&gt;
  * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
  */
-public class AllelicCountCollection {
-    private final List<AllelicCount> counts;
-
-    public AllelicCountCollection() {
-        counts = new ArrayList<>();
+public final class AllelicCountCollection extends SampleLocatableCollection<AllelicCount> {
+    enum AllelicCountTableColumn {
+        CONTIG,
+        POSITION,
+        REF_COUNT,
+        ALT_COUNT,
+        REF_NUCLEOTIDE,
+        ALT_NUCLEOTIDE;
+
+        static final TableColumnCollection COLUMNS = new TableColumnCollection((Object[]) values());
     }
+    
+    private static final Function<DataLine, AllelicCount> ALLELIC_COUNT_DATA_LINE_TO_RECORD_FUNCTION = dataLine -> {
+        final String contig = dataLine.get(AllelicCountTableColumn.CONTIG);
+        final int position = dataLine.getInt(AllelicCountTableColumn.POSITION);
+        final int refReadCount = dataLine.getInt(AllelicCountTableColumn.REF_COUNT);
+        final int altReadCount = dataLine.getInt(AllelicCountTableColumn.ALT_COUNT);
+        final Nucleotide refNucleotide = Nucleotide.valueOf(dataLine.get(AllelicCountTableColumn.REF_NUCLEOTIDE.name()).getBytes()[0]);
+        final Nucleotide altNucleotide = Nucleotide.valueOf(dataLine.get(AllelicCountTableColumn.ALT_NUCLEOTIDE.name()).getBytes()[0]);
+        final SimpleInterval interval = new SimpleInterval(contig, position, position);
+        return new AllelicCount(interval, refReadCount, altReadCount, refNucleotide, altNucleotide);
+    };
+
+    private static final BiConsumer<AllelicCount, DataLine> ALLELIC_COUNT_RECORD_AND_DATA_LINE_BI_CONSUMER = (allelicCount, dataLine) ->
```

Comment:
I like this idea a lot, but in practice it adds quite a bit of boilerplate if we do things using enums.  Also, if we want to do the same thing on the decoding side, then it gets messier.

Perhaps it would be nicer to have interfaces for encoding and decoding columns.  Then we could construct a `SampleRecordCollection` by passing an ordered collection of objects that implement both interfaces.

For now I'll leave things how they are, but I might refactor along these lines before release.